java-extras/src/main/java/org/triplea/java/Retryable.java

```diff
@@ -0,0 +1,113 @@
+package org.triplea.java;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.time.Duration;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+
+/**
+ * Module to execute a task with retries. Provides a builder interface to specify number of max
+ * attempts (max number of times the task will be executed) and backoff.
+ */
+@AllArgsConstructor(access = AccessLevel.PRIVATE)
+public class Retryable {
+  private static final Consumer<Duration> DEFAULT_THREAD_SLEEP =
+      duration -> Interruptibles.sleep(duration.toMillis());
+
+  private final Consumer<Duration> threadSleeper;
+  private final int maxAttempts;
+  private final Duration fixedBackOff;
+  private final Supplier<Boolean> taskRunner;
+
+  public static MaxAttemptsBuilder builder() {
+    return new MaxAttemptsBuilder(DEFAULT_THREAD_SLEEP);
+  }
+
+  @VisibleForTesting
+  static MaxAttemptsBuilder builder(final Consumer<Duration> threadSleeper) {
+    return new MaxAttemptsBuilder(threadSleeper);
+  }
+
+  @AllArgsConstructor(access = AccessLevel.PRIVATE)
+  public static class MaxAttemptsBuilder {
+    private final Consumer<Duration> threadSleeper;
+
+    public BackOffBuilder withMaxAttempts(final int maxAttempts) {
+      Preconditions.checkArgument(
+          maxAttempts > 1,
+          "Max attempt count must be greater than 1, if max attempt is 1, "
+              + "just invoke your task directly without the retry mechanism");
+      return new BackOffBuilder(threadSleeper, maxAttempts);
+    }
+  }
+
+  @AllArgsConstructor(access = AccessLevel.PRIVATE)
+  public static class BackOffBuilder {
+    private final Consumer<Duration> threadSleeper;
+    private final int maxAttempts;
+
+    public TaskBuilder withFixedBackOff(final Duration duration) {
```

Comment:
Hehe, raw numbers for inter-system boundaries (IE: JSON - prefer to emit JSON primitives rather than JSON values). Since we are within a system, using strongest type possible is good (ie: avoid strings and use value objects instead, and generally use strong typing)